app/models/host.rb

```diff
@@ -41,5 +39,21 @@ def filter_by_compliance_score(_filter, operator, score)
 
       { conditions: "hosts.id IN(#{ids.join(',')})" }
     end
+
+    def has_test_results(_filter, operator, value)
+      ids = Host.select do |h|
+        h.test_results.present? == ActiveModel::Type::Boolean.new.cast(value)
+      end
+      ids = ids.pluck(:id).map { |id| "'#{id}'" }
+      scoped_search_sql_conditions(ids, operator)
+    end
+
+    def scoped_search_sql_conditions(ids, operator)
+      return { conditions: '1=0' } if ids.empty?
+      operator = operator == '<>' ? 'NOT' : ''
+      { conditions: "hosts.id #{operator} IN(#{ids.join(',')})" }
```

Comment:
We cannot build queries like this - there is a maximum SQL query limit. I'm not sure if our graphql framework really supports this, but we should be working in ActiveRecord::Relations up until the point of execution against the DB. Arrays , string `join`s, and `map`s are all red flags.